File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int maxLen = 0;
        unordered_set<int> mp(nums.begin(), end(nums));

        for(int num : mp){
            if(mp.count(num-1)) continue;       // if it's previous no is present, no need to process current element.
            int i = num;
            while(mp.count(i+1)) ++i;           // checking next-next no
            maxLen = max(maxLen, (i-num+1));
        }
        return maxLen;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
import java.util.Arrays;
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
        Arrays.sort(nums);
        int mx = 1;
        int cnt = 1;
        for(int i=0; i<nums.length-1; i++)
        {
            if(nums[i] == nums[i+1]) continue;
            else if(nums[i]+1 == nums[i+1])
            {
                cnt++;
            }
            else
            {
                mx = Math.max(cnt, mx);
                cnt = 1;
            }
        }
        mx = Math.max(cnt, mx);
        return mx;
    }
}